test/flows/calculate_agricultural_holiday_entitlement_flow_test.rb

```diff
@@ -0,0 +1,125 @@
+require "test_helper"
+require "support/flow_test_helper"
+
+class CalculateAgriculturalHolidayEntitlementFlowTest < ActiveSupport::TestCase
+  include FlowTestHelper
+
+  setup { testing_flow CalculateAgriculturalHolidayEntitlementFlow }
+
+  should "render a start page" do
+    assert_rendered_start_page
+  end
+
+  context "question: work_the_same_number_of_days_each_week?" do
+    setup { testing_node :work_the_same_number_of_days_each_week? }
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of how_many_days_per_week? for a 'same-number-of-days' response" do
+        assert_next_node :how_many_days_per_week?, for_response: "same-number-of-days"
+      end
+
+      should "have a next node of what_date_does_holiday_start? for a 'different-number-of-days' response" do
+        assert_next_node :what_date_does_holiday_start?, for_response: "different-number-of-days"
+      end
+    end
+  end
+
+  context "question: how_many_days_per_week?" do
+    setup do
+      testing_node :how_many_days_per_week?
+      add_responses work_the_same_number_of_days_each_week?: "same-number-of-days"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of worked_for_same_employer? for any response" do
+        assert_next_node :worked_for_same_employer?, for_response: "7-days"
+      end
+    end
+  end
+
+  context "question: what_date_does_holiday_start?" do
+    setup do
+      testing_node :what_date_does_holiday_start?
+      add_responses work_the_same_number_of_days_each_week?: "different-number-of-days"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of how_many_total_days? for any response in current year" do
+        travel_to Time.zone.parse("2021-01-01")
```

Comment:
```suggestion
      should "have a next node of how_many_total_days? for any response" do
```

I don't think we need the "current year" qualifier because it doesn't do anything if the year is incorrect (these lines seem to do nothing https://github.com/alphagov/smart-answers/blob/f27f74227f27f47dfce636d18b859f453da16577/app/flows/calculate_agricultural_holiday_entitlement_flow.rb#L46-L47 - I guess at one point they set boundaries on select inputs but I can't work out any affect they have now).


Oh and moot point now, but for future, this repo seems to have used timecop for all time travel so it'd be better to stay conventional with that until such a time that we swap that over (_wonders whether it could be done entirely with a find & replace_)

Reply:
Yup, find and replaced away: https://github.com/alphagov/smart-answers/pull/5535